evopy/individual.py

```diff
@@ -38,3 +45,47 @@ def evaluate(self, fitness_function):
         self.fitness = fitness_function(self.genotype)
 
         return self.fitness
+
+    def _reproduce_single_variance(self):
+        """Create a single offspring individual from the set genotype and strategy parameters.
+
+        This function uses the single variance strategy.
+
+        :return: an individual which is the offspring of the current instance
+        """
+        new_genotype = self.genotype + \
+                       self.strategy_parameters[0] * np.random.randn(self.length)
+        scale_factor = np.random.randn() / (2 * self.length)
+        new_strategy = [max(self.strategy_parameters[0] * np.exp(scale_factor), self._EPSILON)]
+        return Individual(new_genotype, self.strategy, new_strategy)
+
+    def _reproduce_multiple_variance(self):
+        """Create a single offspring individual from the set genotype and strategy.
+
+        This function uses the multiple variance strategy.
+
+        :return: an individual which is the offspring of the current instance
+        """
+        new_genotype = self.genotype + [self.strategy_parameters[i] * np.random.randn()
+                                        for i in range(self.length)]
+        global_scale_factor = np.random.randn() / (2 * self.length)
+        scale_factors = [np.random.randn() / 2 * np.sqrt(self.length) for _ in range(self.length)]
+        new_strategy = [max(np.exp(global_scale_factor + scale_factors[i]), self._EPSILON) +
+                        self.strategy_parameters[i] for i in range(self.length)]
+        return Individual(new_genotype, self.strategy, new_strategy)
+
+    def _reproduce_full_variance(self):
+        """Create a single offspring individual from the set genotype and strategy.
+
+        This function uses the full variance strategy.
+
+        :return: an individual which is the offspring of the current instance
+        """
+        new_genotype = self.genotype # TODO: Add rotation
+        global_scale_factor = np.random.randn() / (2 * self.length)
+        scale_factors = [np.random.randn() / 2 * np.sqrt(self.length)]
+        new_variances = [max(np.exp(global_scale_factor + scale_factors[i]), self._EPSILON) +
+                         self.strategy_parameters[i] for i in range(self.length)]
+        new_rotations = [self.strategy_parameters[i] + np.random.randn() * 0.0873
```

Comment:
The slides mention something about a `max` here, not sure what is meant there exactly. We should probably look further into it